### PR TITLE
Do not build regression test-specific targets when not compiling with MFEM

### PR DESCRIPTION
**Pull request by pranav-vempati**
Do not build the libROM regression test targets when the MFEM flag is not present.

### CODE REVIEW

**Comment by chldkdtn** [comment]
@pranav-vempati Thank you for doing this. I have a question. How does this PR affect the CI. Does the CI always compile with MFEM?

**Comment by pranav-vempati** [comment]
@chldkdtn This PR should not affect the CI, as the CI always compiles libROM with MFEM.

**Review by chldkdtn — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -203,42 +203,46 @@ if (ENABLE_EXAMPLES)
...
-endforeach(name) # IN LISTS regression_test_names
+  set(unit_test_names
+    smoke_test
+    test_include
+    uneven_dist
+    weak_scaling
+    random_test
+    smoke_static
+    load_samples)
+  if (USE_MFEM)
```
Comment (siuwuncheung) [inline CMakeLists.txt:214]: Adding a line here may make it clearer.

**Review by siuwuncheung — APPROVED** [review]

**Review by dylan-copeland — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (pranav-vempati) [reply CMakeLists.txt:215]: Fixed.

**Pull request merged**